blob/gcsblob/gcsblob.go

```diff
@@ -94,11 +96,16 @@ type lazyCredsOpener struct {
 
 func (o *lazyCredsOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket, error) {
 	o.init.Do(func() {
-		creds, err := gcp.DefaultCredentials(ctx)
-		if err != nil {
-			o.err = err
-			return
+		creds, _ := google.CredentialsFromJSON(ctx, []byte(`{"type": "service_account", "project_id": "my-project-id"}`))
```

Comment:
Let's do this is an `if/else` to be more clear, something like:

```
var creds google.Credentials // or whatever the type is
if os.Getenv("STORAGE_EMULATOR_HOST") == "" {
  var err error
  creds, err = gcp.DefaultCredentials(ctx)
  if err != nil {
    o.err = err
    return
  }
} else {
  // When using storage emulator, use dummy creds.
  creds, _ = google.CredentialsFromJSON(ctx, []byte(`{"type": "service_account", "project_id": "my-project-id"}`))
}
```

Reply:
Fixed